pkg/core/hostid/aws.go

```diff
@@ -1,51 +1,40 @@
 package hostid
 
 import (
-	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"net/http"
 
-	"github.com/signalfx/signalfx-agent/pkg/utils/timeutil"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
 	log "github.com/sirupsen/logrus"
+
+	"github.com/signalfx/signalfx-agent/pkg/utils/timeutil"
 )
 
 // AWSUniqueID constructs the unique EC2 instance of the underlying host.  If
 // not running on EC2, returns the empty string.
 func AWSUniqueID(cloudMetadataTimeout timeutil.Duration) string {
-	c := http.Client{
+	// Pass in the HTTP client so cloudMetadataTimeout from the config
+	// is respected.
+	c := &http.Client{
 		Timeout: cloudMetadataTimeout.AsDuration(),
 	}
 
-	resp, err := c.Get("http://169.254.169.254/2014-11-05/dynamic/instance-identity/document")
-	if err != nil {
-		log.WithFields(log.Fields{
-			"detail": err,
-		}).Info("No AWS metadata server detected, assuming not on EC2")
-		return ""
-	}
-	defer resp.Body.Close()
-
-	body, err := ioutil.ReadAll(resp.Body)
+	sess, err := session.NewSession(aws.NewConfig().WithHTTPClient(c))
 	if err != nil {
 		log.WithFields(log.Fields{
 			"error": err,
-		}).Info("Failed to read AWS instance-identity response")
+		}).Errorf("Failed to create new session for AWS metadata collection")
 		return ""
 	}
 
-	var doc struct {
-		AccountID  string `json:"accountId"`
-		InstanceID string `json:"instanceId"`
-		Region     string `json:"region"`
-	}
-
-	err = json.Unmarshal(body, &doc)
+	client := ec2metadata.New(sess)
+	doc, err := client.GetInstanceIdentityDocument()
 	if err != nil {
 		log.WithFields(log.Fields{
 			"error": err,
-			"body":  string(body),
-		}).Info("Failed to unmarshal AWS instance-identity response")
+		}).Errorf("Failed to get AWS instance-identity")
```

Comment:
For agents that aren't on EC2, there should only be a debug message saying that they don't appear to be on EC2.  There shouldn't be an error log for people running on non-EC2 platforms, it causes a lot of confusion.

Reply:
I've updated to retain original log levels ptal.